docs/api.md

```diff
@@ -803,7 +803,7 @@ Where:
 
 * array is an Array or promise for an Array, which may contain promises and/or values
 
-Traditional array map function, similar to `Array.prototype.map()`, but allows input to contain promises and/or values, and mapFunc may return either a value or a promise.
+Traditional array map function, similar to `Array.prototype.map()`, but allows input to contain promises and/or values, and mapFunc may return either a value or a promise. The order of items in the input array and the results will match, however, `when.map` allows mapping to proceed opportunistically promises in the array fulfill, making it extremely efficient.
```

Comment:
Trying to figure out what you meant here....  perhaps this?

s/opportunistically promises/opportunistically as promises